src/PIL/PpmImagePlugin.py

```diff
@@ -83,29 +111,22 @@ def _open(self):
             self.mode = rawmode = mode
 
         for ix in range(3):
-            while True:
-                while True:
-                    s = self.fp.read(1)
-                    if s not in b_whitespace:
-                        break
-                    if s == b"":
-                        raise ValueError("File does not extend beyond magic number")
-                if s != b"#":
-                    break
-                s = self.fp.readline()
-            s = int(self._token(s))
-            if ix == 0:
-                xsize = s
-            elif ix == 1:
-                ysize = s
+            token = self._read_token()
+            try:  # check token sanity
+                token = int(token)
+            except ValueError:
+                raise ValueError(f"Non-decimal-ASCII found in header: {token}")
+            if ix == 0:  # token is the x size
+                xsize = token
+            elif ix == 1:  # token is the y size
+                ysize = token
                 if mode == "1":
                     break
-            elif ix == 2:
-                # maxgrey
-                if s > 255:
+            elif ix == 2:  # token is maxval
+                if token > 255:
                     if not mode == "L":
-                        raise ValueError(f"Too many colors for band: {s}")
-                    if s < 2 ** 16:
+                        raise ValueError(f"Too many colors for band: {token}")
```

Comment:
You didn't change this, so you don't have to do this at all, but the code coverage warning here can be fixed by adding a test with an image like this: `b"P6\n1 1\n1000\n"`.

I'd be happy to add that in a separate PR if you don't want to.

Reply:
A test was added for this in https://github.com/python-pillow/Pillow/commit/bc5ecfb79c3dca0553e2b863ba7d306f2e0fd434